modules/zivid/presets.py

```diff
@@ -0,0 +1,144 @@
+"""Contains the Preset and Category classes."""
+import _zivid
+
+from zivid.settings import _to_settings
+from zivid.camera_info import CameraInfo
+
+
+class Preset:
+    """3D settings preset.
+
+    Presets are pre-defined settings that are tuned for different camera models to perform optimally
+    under different conditions and use cases.
+
+    This class cannot be initialized directly by the end-user. Use the presets method on the
+    Category class to obtain a Camera instance.
+    """
+
+    def __init__(self, impl):
+        """Initialize Preset wrapper.
+
+        This constructor is only used internally, and should not be called by the end-user.
+
+        Args:
+            impl:   Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if not isinstance(impl, _zivid.presets.Preset):
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}".format(
+                    type(impl), type(_zivid.presets.Preset)
```

Comment:
`type(_zivid.presets.Preset)` is wrong here (it's always type). You want just `_zivid.presets.Preset`.

Reply:
I would just write "expected zivid.Preset" in the string actually. Something has to be hardcoded anyway, and `_zivid.presets.Preset` inside format() isn't any better.

Reply:
This __init__ function is almost an exact copy of the init functions in `camera.py` and others. I guess they should all be changed. Maybe in a follow up PR?

Reply:
> Something has to be hardcoded anyway

@eskaur If the class gets renamed later no, your string will be quietly wrong but the format version will complain. I vote for using a runtime formatting, the overhead should be negligible.

Reply:
> I guess they should all be changed. Maybe in a follow up PR?
@johningve Yes, probably. But do fix this particular instance before merging, we shouldn't merge it with the bug :)

Reply:
Interestingly, type() does change the output of this error:

With type()

```
TypeError: Unsupported type for argument impl. Got <class 'list'>, expected <class 'pybind11_builtins.pybind11_type'>
```

Without type()
```
TypeError: Unsupported type for argument impl. Got <class 'list'>, expected <class '_zivid._zivid.presets.Category'>
```

I don't know why `_zivid` gets repeated here

Reply:
> > Something has to be hardcoded anyway
> 
> @eskaur If the class gets renamed later on, your string will be quietly wrong but the format version will complain. I vote for using runtime formatting version, the overhead should be negligible.

:+1:

Reply:
Okay I agree :)